book/src/lists.md

```diff
@@ -0,0 +1,210 @@
+As we scroll through our elements, the widgets which become invisible will be reused.
+The following figure demonstrates how this works in practice.
+
+<div style="text-align:center"><img src="img/scalable_lists_concept.png"/></div>
+
+100 000 elements is something `ListBox` will already struggle with, so let us this to demonstrate scalable lists.
+
+We start by defining and filling up our model.
+The model is an instance of [`gio::ListStore`](https://gtk-rs.org/docs/gio/struct.ListStore.html).
+The main limitation here is that `gio::ListStore` only accepts GObjects.
+What we would need is a GObject which holds an integer and exposes it as property.
+To get that we just have to slightly adapt the `CustomButton` we created in the subclassing [chapter](gobject_subclassing.html).
+We only need to let it inherit from GObject instead of `Button` and let the `new` method accept an integer as parameter.
+
+<span class="filename">Filename: listings/lists/2/integer_object/mod.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/2/integer_object/mod.rs:integer_object}}
+#
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+We now fill the model with integers from 0 to 100 000.
+Please note, that the models only takes care of the data.
+Neither `Label` nor any other widget is mentioned here.
+
+<span class="filename">Filename: listings/lists/2/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/2/main.rs:model}}
+```
+
+The `ListItemFactory` takes care of the widgets as well as their relationship to the model.
+Here, we use the [`SignalListItemFactory`](../docs/gtk4/struct.SignalListItemFactory.html) which emits a signal for every relevant step in the life of a [`ListItem`](../docs/gtk4/struct.ListItem.html).
+The "setup" signal will be emitted when new widgets have to be created.
+We connect to it to create a `Label` for every requested widget.
+
+<span class="filename">Filename: listings/lists/2/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/2/main.rs:factory_setup}}
+```
+
+In the "bind" step we bind the data in our model to the individual list items.
+
+<span class="filename">Filename: listings/lists/2/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/2/main.rs:factory_bind}}
+```
+
+We only want single items to be selectable, so we choose [`SingleSelection`](../docs/gtk4/struct.SingleSelection.html).
+The other options would have been [`MultiSelection`](../docs/gtk4/struct.MultiSelection.html) or [`NoSelection`](../docs/gtk4/struct.NoSelection.html).
+Then we pass the model and the factory to the [`ListView`](../git/docs/gtk4/struct.ListView.html).
+
+<span class="filename">Filename: listings/lists/2/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/2/main.rs:selection_list}}
+```
+
+Every `ListView` has to be inside a `ScrolledWindow`, so we are adding it to one.
```

Comment:
Maybe explicitly mention that it has to be a direct child of a scrolledwindow, not inside a viewport or a box in-between? it's a common mistake.